components/fxa-client/src/internal/auth_state.rs

```diff
@@ -0,0 +1,213 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::collections::{HashMap, HashSet};
+
+use crate::{
+    internal::{
+        commands, device,
+        oauth::{AccessTokenInfo, RefreshToken},
+        profile::Profile,
+        state_persistence::state_to_json,
+        CachedResponse, Config, OAuthFlow, PersistedState,
+    },
+    Result, ScopedKey,
+};
+
+/// Stores the current state of the FxA client
+///
+/// In order to make reasoning about state changes easier, all fields are private, and all state
+/// changes are confined to this module.
+pub struct AuthState {
+    /// State that's persisted to disk, which is the majority of the auth state
+    persisted_state: PersistedState,
+    /// In-progress OAuth flows
+    flow_store: HashMap<String, OAuthFlow>,
```

Comment:
There's not many fields in this new struct.  I was tempted to just make the fields in `PersistedState` private and add the functionality below to that struct directly.  However, I think there's a good chance we'll want to add more fields in the future, for example `AuthState` could eventually store a `StorageHandler` object and have any `&mut` methods call `StorageHandler.save_state()`.  That also a good example of why I think it's nice to put all the state mutations in the same place.